core/math/a_star.cpp

```diff
@@ -47,7 +47,7 @@ int64_t AStar3D::get_available_point_id() const {
 
 void AStar3D::add_point(int64_t p_id, const Vector3 &p_pos, real_t p_weight_scale) {
 	ERR_FAIL_COND_MSG(p_id < 0, vformat("Can't add a point with negative id: %d.", p_id));
-	ERR_FAIL_COND_MSG(p_weight_scale < 0.0, vformat("Can't add a point with weight scale less than 0.0: %f.", p_weight_scale));
+	ERR_FAIL_COND_MSG(p_weight_scale < 0.0_R, vformat("Can't add a point with weight scale less than 0.0: %f.", p_weight_scale));
```

Comment:
That previous usage looks super incorrect, because afaik it causes promotion of 32 bit `p_weight_scale` to double. If not using this `_R`, we should either be using `0` or `0.0f`, both of which work fine in 32 bit, and will be promoted to double if `real_t` is double.

Reply:
The return of `_R` will be real_t, `p_weight_scale` being real_t and the return of any `_R` being real_t, I feel it shouldn't require any promotion.

Reply:
lawnjelly is talking about the *previous* behavior, where it was a double regardless of the precision configuration